third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -543,6 +544,55 @@ impl ValueImpl {
 
         Ok(res)
     }
+
+    fn compare(&self, other: &Self) -> PartialVMResult<Ordering> {
+        use ValueImpl::*;
+
+        let res = match (self, other) {
+            (U8(l), U8(r)) => l.cmp(r),
+            (U16(l), U16(r)) => l.cmp(r),
+            (U32(l), U32(r)) => l.cmp(r),
+            (U64(l), U64(r)) => l.cmp(r),
+            (U128(l), U128(r)) => l.cmp(r),
+            (U256(l), U256(r)) => l.cmp(r),
+            (Bool(l), Bool(r)) => l.cmp(r),
+            (Address(l), Address(r)) => l.cmp(r),
+
+            (Container(l), Container(r)) => l.compare(r)?,
+
+            (ContainerRef(l), ContainerRef(r)) => l.compare(r)?,
+            (IndexedRef(l), IndexedRef(r)) => l.compare(r)?,
+
+            // Disallow equality for delayed values. The rationale behind this
+            // semantics is that identifiers might not be deterministic, and
+            // therefore equality can have different outcomes on different nodes
+            // of the network. Note that the error returned here is not an
+            // invariant violation but a runtime error.
```

Comment:
Does equality already do the same? 

I wonder whether this should use the same error code as equality, which definitely is not VM_EXTENSION__ERROR.

Reply:
Comment copy-pasted from fn equals. Let's at least change Disallow equality to Disallow comparison. I'd also remove rest of the comment and just write (see ValueImpl::fn equals).

@wrwg The VM_EXTENSION_ERROR makes sense, it's an error of the aggregators extension.